nx/lib/nx/binary_backend.ex

```diff
@@ -1115,15 +1117,52 @@ defmodule Nx.BinaryBackend do
   def triangular_solve(
         %{type: output_type} = out,
         %{type: a_type, shape: {rows, rows}} = a,
-        %{type: b_type, shape: {rows}} = b,
+        %{type: b_type, shape: b_shape} = b,
         opts
-      ) do
+      )
+      when b_shape == {rows, rows} or b_shape == {rows} do
     a_data = to_binary(a)
     b_data = to_binary(b)
-    out_bin = B.Decomposition.ts(a_data, a_type, b_data, b_type, rows, output_type, opts)
+    out_bin = B.Decomposition.ts(a_data, a_type, b_data, b_type, b_shape, output_type, opts)
     from_binary(out, out_bin)
   end
 
+  @doc false
+  def solve(%T{type: output_type} = output_holder, %T{shape: {n, n}} = a, %T{shape: b_shape} = b)
+      when b_shape == {n, n} or b_shape == {n} do
+    # We need to achieve an LQ decomposition for `a` (henceforth called A)
+    # because triangular_solve only accepts lower_triangular matrices
+    # Therefore, we can use the fact that if we have M = Q'R' -> transpose(M) = LQ.
+    # If we set M = transpose(A), we can get A = LQ through transpose(qr(transpose(A)))
+
+    # Given A = LQ, we can then solve AX = B as shown below
+    # AX = B -> L(QX) = B -> LY = B, where Y = QX -> X = transpose(Q)Y
+
+    # Finally, it can be shown that when B is a matrix, X can be
+    # calculated by solving for each corresponding column in B
+    transpose_axes = Nx.Shape.transpose_axes({n, n})
+    a_transposed = transpose(a, a, transpose_axes)
+
+    {qt, r_prime} =
+      qr(
+        {%{a_transposed | type: output_type}, %{a_transposed | type: output_type}},
+        a_transposed,
+        []
+      )
+
+    l = transpose(r_prime, r_prime, transpose_axes)
+
+    y = triangular_solve(l, l, b, [])
+
+    dot(output_holder, qt, [1], y, [0])
+  end
+
+  @impl true
+  def invert(output, %T{shape: {n, n}} = tensor) do
+    identity = eye(tensor, [])
+    solve(output, tensor, identity)
+  end
```

Comment:
Given both solve and invert work exclusively on existing operations, I think we can move this implementations to Nx. WDYT? This way we don't add any new function to the backend. :)

Reply:
For this function specifically I think we can benefit of existing implementations on XLA and other backends. Matrix inversion has lots of optimized versions (with divide and conquer algorithms and such), so I'd rather delegate it to the low-level backends whenever possible